### PR TITLE
Fix certificate information extraction from the response (#769)

### PR DESCRIPTION
**Pull request by leviliangtw**
## Issue

- #769

## Changes

- [X] Add a new class `cpr::CertInfo`
- [X] Rename the `std::vector<std::string> Response::GetCertInfo()` to `std::vector<CertInfo> Response::GetCertInfos()` and rewrite it
- [X] Rename the unit test `SslTests.GetCertInfo` to `SslTests.GetCertInfos` and rewrite it

## New Class: `cpr::CertInfo`

`cpr::CertInfo` is a class of a vector of `std::string`, and the string vector stores kinds of information about a certificate: 

```cpp
class CertInfo {
  private:
    std::vector<std::string> cert_info_;

  public:
    CertInfo() = default;
    CertInfo(const std::initializer_list<std::string>& entry) : cert_info_{entry} {};
    ~CertInfo() noexcept = default;

    using iterator = std::vector<std::string>::iterator;
    using const_iterator = std::vector<std::string>::const_iterator;

    std::string& operator[](const size_t& pos);
    iterator begin();
    iterator end();
    const_iterator begin() const;
    const_iterator end() const;
    const_iterator cbegin() const;
    const_iterator cend() const;
    void emplace_back(const std::string& __x);
    void push_back(const std::string& __x);
    void pop_back();
};
```

## Fixed Method: `std::vector<CertInfo> Response::GetCertInfos()` 

The original method is named `std::vector<std::string> Response::GetCertInfo()`; however, the name does not match its expected behavior. 
This renamed method is fixed according to the correct usage in the doc of `libcurl` - [CURLINFO_CERTINFO](https://curl.se/libcurl/c/CURLINFO_CERTINFO.html) - and you could simply iterate the  `certInfos` like that: 

```cpp
......
Response response = cpr::Get(url, sslOpts, Timeout{5000}, Verbose{});

std::vector<CertInfo> certInfos = response.GetCertInfos();
for (const auto& certinfo : certInfos) {
    for (const auto& entry : certinfo) {
        std::cout << entry << ", ";
    }
}
```

## Unit Test: `SslTests.GetCertInfos`

To make the unit test resonable, I created the expected variant `expectedCertInfos` for assertion just like the following example: 

```cpp
std::vector<CertInfo> expectedCertInfos{
        CertInfo{
                "Subject:CN = test-server",
                "Issuer:C = GB, O = Example, CN = Root CA",
                "Version:2",
                "Serial Number:28c252871ec62a626a98006b0bf2888f",
                "Signature Algorithm:ED25519",
                "Public Key Algorithm:ED25519",
                "X509v3 Subject Alternative Name:DNS:localhost, IP Address:127.0.0.1, IP Address:0:0:0:0:0:0:0:1",
                "X509v3 Subject Key Identifier:39:C1:81:38:01:DC:55:38:E5:2F:4E:7A:D0:4C:84:7B:B7:27:D3:AF",
                "X509v3 Authority Key Identifier:keyid:E4:F2:F3:85:0E:B7:85:75:84:76:E3:43:D1:B6:9D:14:B8:E2:A4:B7\n",
                "Start date:Jun 29 11:33:07 2022 GMT",
                "Expire date:Jun 28 11:33:07 2027 GMT",
                "Signature:2e:0d:a1:0d:f5:90:77:e9:eb:84:7d:80:63:63:4d:8a:eb:d9:23:57:1f:21:2a:ed:81:b4:a8:58:b9:00:1b:cb:5c:90:1b:33:6b:f6:ec:42:20:63:54:d6:60:ee:37:14:1b:1c:95:0b:33:ea:67:29:d4:cc:d9:7e:34:fd:47:04:",
                R"(Cert:-----BEGIN CERTIFICATE-----
MIIBdTCCASegAwIBAgIQKMJShx7GKmJqmABrC/KIjzAFBgMrZXAwMTELMAkGA1UE
BhMCR0IxEDAOBgNVBAoMB0V4YW1wbGUxEDAOBgNVBAMMB1Jvb3QgQ0EwHhcNMjIw
NjI5MTEzMzA3WhcNMjcwNjI4MTEzMzA3WjAWMRQwEgYDVQQDDAt0ZXN0LXNlcnZl
cjAqMAUGAytlcAMhAI64JU5RjfdEG1KQMxS5DQWkiGlKIQO7ye4mNFq9QleTo3Aw
bjAsBgNVHREEJTAjgglsb2NhbGhvc3SHBH8AAAGHEAAAAAAAAAAAAAAAAAAAAAEw
HQYDVR0OBBYEFDnBgTgB3FU45S9OetBMhHu3J9OvMB8GA1UdIwQYMBaAFOTy84UO
t4V1hHbjQ9G2nRS44qS3MAUGAytlcANBAC4NoQ31kHfp64R9gGNjTYrr2SNXHyEq
7YG0qFi5ABvLXJAbM2v27EIgY1TWYO43FBsclQsz6mcp1MzZfjT9RwQ=
-----END CERTIFICATE-----
)",
        },
};
```

### CODE REVIEW

File: include/cpr/response.h [hunk 1]
```diff
@@ -16,6 +16,30 @@
...
+    using const_iterator = std::vector<std::string>::const_iterator;
+
+    std::string& operator[](const size_t& pos);
+    iterator begin();
+    iterator end();
+    const_iterator begin() const;
+    const_iterator end() const;
+    const_iterator cbegin() const;
+    const_iterator cend() const;
+    void emplace_back(const std::string& __x);
```
Comment (COM8) [inline include/cpr/response.h:38]: Clang Tidy says: `Declaration uses identifier '__x', which is a reserved identifier`

Also we should use something more descriptive here like `str`.

File: include/cpr/response.h [hunk 2]
```diff
@@ -16,6 +16,30 @@
 
 namespace cpr {
 
+class CertInfo {
```
Comment (COM8) [inline include/cpr/response.h:19]: Please extract `CertInfo` into its own file.

**Review by COM8 — CHANGES_REQUESTED** [review]
Looks good. Thanks!
Please also create a PR to update the docs:
https://docs.libcpr.org/advanced-usage.html#response-objects
https://docs.libcpr.org/advanced-usage.html#https-options

File: include/cpr/response.h (hunk 1 above)
Comment (leviliangtw) [reply include/cpr/response.h:38]: Fixed it!

File: include/cpr/response.h (hunk 2 above)
Comment (leviliangtw) [reply include/cpr/response.h:19]: No problem!

**Comment by leviliangtw** [comment]
> Looks good. Thanks! Please also create a PR to update the docs: https://docs.libcpr.org/advanced-usage.html#response-objects https://docs.libcpr.org/advanced-usage.html#https-options

The docs PR [#27](https://github.com/libcpr/docs/pull/27) is ready.

File: include/cpr/cert_info.h [hunk 3]
```diff
@@ -0,0 +1,45 @@
+#ifndef CPR_CERT_INFO_H
+#define CPR_CERT_INFO_H
+
+// #include <cassert>
```
Comment (COM8) [inline include/cpr/cert_info.h:4]: Commented out stuff can be removed.

**Review by COM8 — CHANGES_REQUESTED** [review]
Please add `cert_info.h` into `cpr.h`.

**Comment by COM8** [comment]
Fixed the Windows CI issue: https://github.com/libcpr/cpr/commit/00fe6e948c2263201765816354551f6846b8c8f5

**Review by COM8 — APPROVED** [review]

**Pull request merged**